### DISCUSSION TITLE
status 200 at dev mode and status 500 at production mode

### DISCUSSION THREAD

**Discussion by dracoalv** [vercel/next.js#39444] [Help]
this code are fine in development mode, but in production mode, return a status 500.

```jsx
import type { NextApiRequest, NextApiResponse } from 'next'
import { getGoogleMetricsData } from '~/lib/firebase/analytics/data';

export default async function handler(req: NextApiRequest,res: NextApiResponse) {
  try {
    const gaData = await getGoogleMetricsData()
    res.status(200).json({data: gaData}.data)
  } catch (error) {
    console.log(error)
    res.status(500).send({ error: 'failed to fetch data' } as any)
  }
}
```
dev mode [localhost]
![image](https://user-images.githubusercontent.com/70287096/183809582-294cac71-05e8-495f-8f2f-2ad1980ae5bc.png)

prod mode [vercel host]
![image](https://user-images.githubusercontent.com/70287096/183809725-ffe77a35-1451-47c9-8d7d-c0ec02972949.png)

# About project
In this code i use firebase services including google analytics data to create my own dashboard, but its only works in dev mode but i dont know why

**Comment by gaston-flores** [comment]
Hey @dracoalv, can you share your `getStaticProps` or `getServerSideProps` function for `dashboard`? It looks like the SSR is failing for that page. Have you looked at the logs in Vercel?
